packages/compass-indexes/src/modules/regular-indexes.ts

```diff
@@ -178,125 +210,140 @@ export default function reducer(state = INITIAL_STATE, action: AnyAction) {
   return state;
 }
 
-export const setRegularIndexes = (
-  indexes: RegularIndex[]
-): IndexesAddedAction => ({
-  type: ActionTypes.IndexesAdded,
-  indexes,
+const fetchIndexesStarted = (
+  isRefreshing: boolean
+): FetchIndexesStartedAction => ({
+  type: ActionTypes.FetchIndexesStarted,
+  isRefreshing,
 });
 
-const setIsRefreshing = (isRefreshing: boolean): SetIsRefreshingAction => ({
-  type: ActionTypes.SetIsRefreshing,
-  isRefreshing,
+const fetchIndexesSucceeded = (
+  indexes: RegularIndex[]
+): FetchIndexesSucceededAction => ({
+  type: ActionTypes.FetchIndexesSucceeded,
+  indexes,
 });
 
-const setError = (error: string | null): SetErrorAction => ({
-  type: ActionTypes.SetError,
+const fetchIndexesFailed = (error: string): FetchIndexesFailedAction => ({
+  type: ActionTypes.FetchIndexesFailed,
   error,
 });
 
-const _handleIndexesChanged = (
-  indexes: RegularIndex[]
-): IndexesThunkAction<void> => {
-  return (dispatch, _, { localAppRegistry }) => {
-    dispatch(setRegularIndexes(indexes));
-    dispatch(setIsRefreshing(false));
-    localAppRegistry?.emit('indexes-changed', indexes);
-  };
-};
+type FetchIndexesActions =
+  | FetchIndexesStartedAction
+  | FetchIndexesSucceededAction
+  | FetchIndexesFailedAction;
 
-export const fetchIndexes = (): IndexesThunkAction<
-  Promise<void>,
-  RegularIndexesActions
-> => {
-  return async (dispatch, getState, { dataService }) => {
-    const {
-      isReadonlyView,
-      namespace,
-      regularIndexes: { inProgressIndexes },
-    } = getState();
+export const fetchIndexes = (
+  isRefreshing = false
+): IndexesThunkAction<Promise<void>, FetchIndexesActions> => {
+  return async (dispatch, getState, { dataService, localAppRegistry }) => {
+    const { isReadonlyView, namespace } = getState();
 
     if (isReadonlyView) {
-      dispatch(_handleIndexesChanged([]));
+      dispatch(fetchIndexesSucceeded([]));
       return;
     }
 
     try {
-      dispatch(setError(null));
+      dispatch(fetchIndexesStarted(isRefreshing));
+      // The number in the header could go up or down whenever an index is added
+      // or removed.
+      localAppRegistry.emit('refresh-collection-stats');
```

Comment:
Super nit: took me a moment to get why we're doing it on fetch start and not later, maybe we can add something a bit more detailed, along the lines of "this makes sure that when user or something else triggers a re-fetch for the list of indexes with this action, the tab header also gets updated"